packages/ai/src/LLMStepInstance.ts

```diff
@@ -100,6 +103,12 @@ export class LLMStepInstance<
     });
   }
 
+  instanceOf<TemplateShape extends IOShape>(
+    template: LLMStepTemplate<TemplateShape>
+  ): this is LLMStepInstance<TemplateShape, WorkflowShape> {
```

Comment:
I'm happy that this worked, no need for registry (which would be messy to implement) to narrow the type.